### ISSUE TITLE
Bug: [no-unused-expressions] expression with type arguments is not flagged

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#7748]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.4&fileType=.tsx&code=MYGwhgzhAEBiD28A8AVAfNA3tAvgWAChCB6Y6CAC3gFcQATaAIwFNoBbMAJwGtmHJo1AHbUIfQgmQQALpwCWQgOZpChAIwBuQkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAm9AB7FoKZJQ5NUGSImjQO0SODABfEOqA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```typescript
class Foo<T> { }

// should be marked as unused
Foo<string>
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unused-expressions": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

Expected 1 error as the expression-with-type-arguments is a pure expression.

### Actual Result

No errors.
